react/week2/todolist/src/components/Todo.js

```diff
@@ -0,0 +1,44 @@
+import React, { useState } from "react";
+import TodoForm from "./TodoForm";
+import TodoLists from "./TodoLists";
+import { RiCloseCircleLine } from "react-icons/ri";
+import { TiEdit } from "react-icons/ti";
+
+function Todo({ todos, completeTodo, removeTodo, updateTodo }) {
+  const [edit, setEdit] = useState({
+    id: null,
+    value: "",
+  });
+
+  const submitUpdate = (value) => {
+    updateTodo(edit.id, value);
+    setEdit({ id: null, value: "" });
+  };
+
+  if (edit.id) {
+    return <TodoForm edit={edit} onSubmit={submitUpdate} />;
+  }
+
+  return todos.map((todo, index) => (
+    <div
+      className={todo.isComplete ? "todo-row complete" : "todo-row"}
+      key={index}
+    >
+      <div key={todo.id} onClick={() => completeTodo(todo.id)}>
+        {todo.text}
+      </div>
+      <div className="icons">
+        <RiCloseCircleLine
+          onClick={() => removeTodo(todo.id)}
+          className="delete-icon"
+        />
+        <TiEdit
+          onClick={() => setEdit({ id: todo.id, value: todo.text })}
+          className="edit-icon"
+        />
+      </div>
+    </div>

```

Comment:
Check my other comment in the TodoList.js.
This is the toDo component (singular), so it makes more sense if it has a single todo in its props and not the entire list.
The TodoList is more appropriate to map the todos!